torch/fx/_symbolic_trace.py

```diff
@@ -809,30 +810,49 @@ def forward(*args, **kwargs):
                 )
                 return self.call_module(mod, forward, args, kwargs)
 
-            with _new_patcher() as patcher:
-                # allow duplicate patches to support the case of nested calls
-                patcher.patch_method(
-                    torch.nn.Module,
-                    "__getattr__",
-                    module_getattr_wrapper,
-                    deduplicate=False,
-                )
-                patcher.patch_method(
-                    torch.nn.Module, "__call__", module_call_wrapper, deduplicate=False
-                )
-                _patch_wrapped_functions(patcher)
-                _autowrap_check(patcher, fn_globals, self._autowrap_function_ids)
-                for module in self._autowrap_search:
-                    _autowrap_check(
-                        patcher, module.__dict__, self._autowrap_function_ids
+            try:
+                with _new_patcher() as patcher:
+                    # allow duplicate patches to support the case of nested calls
+                    patcher.patch_method(
+                        torch.nn.Module,
+                        "__getattr__",
+                        module_getattr_wrapper,
+                        deduplicate=False,
                     )
-                self.create_node(
-                    "output",
-                    "output",
-                    (self.create_arg(fn(*args)),),
-                    {},
-                    type_expr=fn.__annotations__.get("return", None),
-                )
+                    patcher.patch_method(
+                        torch.nn.Module,
+                        "__call__",
+                        module_call_wrapper,
+                        deduplicate=False,
+                    )
+                    _patch_wrapped_functions(patcher)
+                    _autowrap_check(patcher, fn_globals, self._autowrap_function_ids)
+                    for module in self._autowrap_search:
+                        _autowrap_check(
+                            patcher, module.__dict__, self._autowrap_function_ids
+                        )
+                    self.create_node(
+                        "output",
+                        "output",
+                        (self.create_arg(fn(*args)),),
+                        {},
+                        type_expr=fn.__annotations__.get("return", None),
+                    )
+            except RuntimeError as e:
```

Comment:
nit: there's an outer `try/finally` that starts on line 733, should we just an except to that instead of nesting an inner `try/except` block?